Tools/cases_generator/opcode_metadata_generator.py

```diff
@@ -107,6 +108,96 @@ def add(inst: Instruction | PseudoInstruction) -> None:
     emit_stack_effect_function(out, "popped", sorted(popped_data))
     emit_stack_effect_function(out, "pushed", sorted(pushed_data))
 
+    generate_max_stack_effect_function(analysis, out)
+
+
+def emit_max_stack_effect_function(
+    out: CWriter, effects: list[tuple[str, list[str]]]
```

Comment:
We already have code for these sequences of values, that supports simplification: https://github.com/python/cpython/blob/main/Tools/cases_generator/stack.py#L111
It should support this use case without too much modification.

That way you can avoid this: 
```C
        case COPY: {
            *effect = (-1 - (oparg-1)) + (2 + (oparg-1));
```
and have this:
```C
        case COPY: {
            *effect = 1;
```